components/auth/join/join.ts

```diff
@@ -31,11 +41,15 @@ export class AppAuthJoin extends Vue {
 		UserLinkedAccounts.login(this.$router, provider);
 	}
 
-	onJoined(formModel: any) {
-		this.$emit('joined', formModel);
+	onJoining(formModel: FormModel) {
+		this.$emit('joining', formModel);
 
 		if (this.shouldRedirect) {
-			window.location.href = Environment.authBaseUrl + '/join/almost';
+			let url = Environment.authBaseUrl + '/join/almost';
```

Comment:
Change the join-almost route to access the token from session storage instead of a url param.